cpp/arcticdb/processing/clause.cpp

```diff
@@ -304,6 +304,19 @@ Composite<ProcessingUnit> AggregationClause::process(std::shared_ptr<Store> stor
                                                 // 11.01 seconds with caching
                                                 // Not worth worrying about right now
                                                 robin_hood::unordered_flat_map<RawType, size_t> offset_to_group;
+
+                                                bool is_sparse = col.column_->is_sparse();
+                                                using optional_iter_type = std::optional<decltype(input_data.bit_vector()->first())>;
+                                                optional_iter_type iter = std::nullopt;
+                                                size_t previous_value_index = 0;
+
+                                                if (is_sparse)
+                                                {
+                                                    iter = std::make_optional(input_data.bit_vector()->first());
+                                                    // We use 0 for the missing value group id
+                                                    next_group_id++;
```

Comment:
This LGTM, but I cannot tell if this can invalidate some potential implicit invariant the aggregation clause or its tests might have been relying one.